starter-code/main.js

```diff
@@ -0,0 +1,146 @@
+    printMenu();
+    rl.question('Type an option: ', (option) => {
+      switch (option) {
+      case '1':
+        db.collection('companies').find({}, {name: 1, _id: 0}).toArray((error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '2':
+        db.collection('companies').count((error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '3':
+        db.collection('companies').count({founded_year: 2004}, (error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '4':
+        db.collection('companies').find({founded_year: 2004, founded_month: 2}, {name: 1, _id: 0}).toArray((error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '5':
+        db.collection('companies').find(
+          {founded_year: 2004, founded_month: { $gt: 3, $lt: 7 } }, {name: 1, _id: 0}
+        ).sort(
+          {founded_day: 1, founded_month: 1}
+        ).toArray((error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '6':
+        db.collection('companies').find({'offices': {$elemMatch: {city: 'Barcelona'}}}, {name: 1, _id: 0}).toArray((error, result) => {
+          if (error) {
+            console.log(error);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          } else {
+            console.log(result);
+            rl.question(`\nType enter to continue: `, (answer) => { mainMenu(); });
+          }
+        });
+        break;
+      case '7':
+        db.collection('companies').find({}, {name: 1, _id: 0}, {name: 1, _id: 0}).limit(10).sort(
```

Comment:
The order matters! First you need to sort, then you can limit. Your query is going to take 10 companies (not the ones with more employees) and then order them by number of employees.